lib/node_modules/@stdlib/repl/lib/main.js

```diff
@@ -268,6 +280,72 @@ function REPL( options ) {
 		}
 	}
 
+	/**
+	* Callback invoked upon a readline interface "keypress" event.
+	*
+	* @private
+	* @param {string} chunk - character pressed by the user
+	* @param {Object} key - keypress event information
+	*/
+	function onKeypress( chunk, key ) {
+		var charsAfterCursor;
+		var codeAfterCursor;
+		var symbols = {
+			'{': '}',
+			'[': ']',
+			'(': ')',
+			'\'': '\'',
+			'"': '"',
+			'`': '`'
+		};
+
+		// For ALT+ENTER keypress, manually enter multi-line mode:
+		if ( key && key.name === 'return' && key.meta ) {
+			charsAfterCursor = self._rli.line.length - self._rli.cursor;
+			codeAfterCursor = self._rli.line.substring( self._rli.cursor );
+
+			// Update flag
+			self._multiline_mode.manual = true;
+
+			// Store code after cursor in buffer for next line
+			self._multiline_mode.buffer = codeAfterCursor;
+			self._ostream.write( repeat( '\x1b[C', charsAfterCursor ) + repeat( '\b \b', charsAfterCursor ) );
+			self._rli.line = self._rli.line.substring( 0, self._rli.cursor );
+
+			// Simulate `line` event
+			self._rli.write( null, {
+				'name': 'return'
+			});
+		}
+
+		// For a bracket/quotation input, append corresponding closing symbol
+		if ( self._settings.autoCloseSymbols && symbols[chunk] ) {
+			self._rli.write( symbols[chunk] );
+			self._rli.write( null, {
+				'ctrl': true,
+				'name': 'b'
+			} );
+			self._auto_close_status = true;
+			return;
+		}
+
+		// Handle instinctive closing symbols
+		if ( self._auto_close_status ) {
+			if (
+				typeof chunk !== 'undefined' &&
+				symbols[ self._rli.line[ self._rli.cursor - 2 ] ] === chunk
+			) {
+				self._rli.write( null, {
+					'name': 'backspace'
+				} );
+				self._rli.write( null, {
+					'name': 'right'
+				} );
+			}
+			self._auto_close_status = false;
+		}
+	}
```

Comment:
@kgryte should I refactor and move this logic out to a separate file?

Reply:
I'd move to a separate function and make a factory function, where the parent function takes in a REPL instance, creating a closure, and returns this event listener.

Reply:
You can start off with a common file for handling all keypress events, although we may need to be cognizant of https://github.com/stdlib-js/stdlib/pull/1832. In which case, refactoring to a common file could be done in a separate follow-up PR.